Gems/Multiplayer/Code/Include/Multiplayer/Components/NetworkRigidBodyComponent.h

```diff
@@ -61,7 +61,9 @@ namespace Multiplayer
         NetworkRigidBodyComponentController(NetworkRigidBodyComponent& parent);
         void OnActivate(Multiplayer::EntityIsMigrating entityIsMigrating) override;
         void OnDeactivate(Multiplayer::EntityIsMigrating entityIsMigrating) override;
+#if AZ_TRAIT_SERVER_ENABLED
         void HandleSendApplyImpulse(AzNetworking::IConnection* invokingConnection, const AZ::Vector3& impulse, const AZ::Vector3& worldPoint) override;
+#endif
```

Comment:
Something such as this `#if` block technically doesn't need to be in the header.
The #if statement only needs to be in the cpp file.

This function can even be deleted within the Client .cpp file to make it explicit to the user that this function isn't implemented.

```
#if AZ_TRAIT_SERVER_ENABLED
    void NetworkRigidBodyComponentController::HandleSendApplyImpulse(AzNetworking::IConnection* invokingConnection, const AZ::Vector3& impulse, const AZ::Vector3& worldPoint)
    {
       // server implementation
    }
#else
    // Only the server can send impulse to a rigid body
    void NetworkRigidBodyComponentController::HandleSendApplyImpulse(AzNetworking::IConnection* invokingConnection, const AZ::Vector3& impulse, const AZ::Vector3& worldPoint) = delete;
#endif
```

Reply:
I'll look at querying the compile defs from jinja to do this next. It would definitely be a lot clearer.

Reply:
No, I wouldn't do that, if I understand what you're thinking about correctly..

I would keep the separation between jinja and C++ compilation total and complete, as it is now. Trying to route preprocessor definitions into jinja just for this one use-case (which can totally be implemented without that coupling) seems really confusing and error prone. For an end customer, what's the difference between a preprocessor macro passed through cmake vs one defined in a header file somewhere. We'd have to clearly explain concepts like this to our customers so they understand why some preprocessor values work in their jinja code and others seem to be lost or forgotten. What about changing between debug and profile and release builds? this would change the preprocessor macros, so we'd have to regenerate all autogen outputs which would cause all autogen files to rebuild whenever you switch settings like this. What about different projects that use different preprocessor macros, but link a static lib containing autogen code compiled in a different project? Those are just a few edge cases I can think of off the top of my head. It's surprising behaviour, and it could complicate the build pipeline a lot.